modules/dnn/test/test_tf_importer.cpp

```diff
@@ -1571,11 +1571,23 @@ TEST_P(Test_TensorFlow_layers, tf2_permute_nhwc_ncwh)
     runTensorFlowNet("tf2_permute_nhwc_ncwh");
 }
 
-// issue #21852
 TEST_P(Test_TensorFlow_layers, tf_graph_simplifier_buffer_overflow)
```

Comment:
No need to parameterize this test through execution backends. This is an importer only problem

```
TEST(Test_TensorFlow_Importer, tf_graph_simplifier_buffer_overflow_XXXXX)
```

Please split this test into two: `_21852` and `_21947`. No need to merge tests into one - it is hard to debug them (need to skip unrelated code which is annoying).

---

Move test code to the end of this file (it is not a layer test).